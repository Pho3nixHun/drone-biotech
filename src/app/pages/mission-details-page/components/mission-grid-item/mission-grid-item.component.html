<section class="bg-white rounded-lg p-5 space-y-5 w-full shadow-md">
    <h3 class="text-lg font-medium tracking-wide">
        {{ vm().titleKey | transloco }}
    </h3>
    <div class="space-y-5 h-full">
        <ng-content />
    </div>
</section>
