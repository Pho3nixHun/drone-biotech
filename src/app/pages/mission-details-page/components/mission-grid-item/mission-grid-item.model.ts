import { WithTitle } from '@interfaces/with-title.interface';

export type MissionGridItemVM = WithTitle;
