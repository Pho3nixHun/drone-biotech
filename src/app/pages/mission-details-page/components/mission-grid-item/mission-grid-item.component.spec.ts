import { ComponentFixture, TestBed } from '@angular/core/testing';

import { MissionGridItemComponent } from './mission-grid-item.component';
import { getTranslocoModule } from 'transloco-testing.module';
import { Component, input } from '@angular/core';
import { MissionGridItemVM } from './mission-grid-item.model';

const en = { title: 'val' };
const mockVm = { titleKey: en.title };

@Component({
    imports: [MissionGridItemComponent],
    template: `
        <app-mission-grid-item [vm]="vm()">
            <div>Should be projected</div>
        </app-mission-grid-item>
    `,
})
class TestHostComponent {
    public vm = input.required<MissionGridItemVM>();
}
describe('MissionGridItemComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                TestHostComponent,
                getTranslocoModule({
                    langs: { en: en },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template if the input is provided and project the content', () => {
        // Arrange
        fixture.componentRef.setInput('vm', mockVm);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
