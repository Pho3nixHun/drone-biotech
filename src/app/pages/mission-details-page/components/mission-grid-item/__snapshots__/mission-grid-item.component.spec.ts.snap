// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionGridItemComponent should render the template if the input is provided and project the content 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-mission-grid-item
    ng-reflect-vm="[object Object]"
  >
    <section
      class="bg-white rounded-lg p-5 space-y-5 w-full shadow-md"
    >
      <h3
        class="text-lg font-medium tracking-wide"
      >
         val 
      </h3>
      <div
        class="space-y-5 h-full"
      >
        <div>
          Should be projected
        </div>
      </div>
    </section>
  </app-mission-grid-item>
</div>
`;
